Cubical/Foundations/SIP.agda

```diff
@@ -24,115 +24,93 @@ open import Cubical.Foundations.Structure public
   variable
-    ℓ ℓ' ℓ'' ℓ''' : Level
     ℓ₁ ℓ₂ ℓ₃ ℓ₄ ℓ₅ : Level
-    S : Type ℓ → Type ℓ'
+    S : Type ℓ₁ → Type ℓ₂
 
 -- For technical reasons we reprove ua-pathToEquiv using the
 -- particular proof constructed by iso→HAEquiv. The reason is that we
 -- want to later be able to extract
 --
 --   eq : ua-au (ua e) ≡ cong ua (au-ua e)
 --
-uaHAEquiv : (A B : Type ℓ) → HAEquiv (A ≃ B) (A ≡ B)
+uaHAEquiv : (A B : Type ℓ₁) → HAEquiv (A ≃ B) (A ≡ B)
 uaHAEquiv A B = iso→HAEquiv (iso ua pathToEquiv ua-pathToEquiv' pathToEquiv-ua)
 open isHAEquiv
 
 -- We now extract the particular proof constructed from iso→HAEquiv
 -- for reasons explained above.
-ua-pathToEquiv : {A B : Type ℓ} (e : A ≡ B) → ua (pathToEquiv e) ≡ e
+ua-pathToEquiv : {A B : Type ℓ₁} (e : A ≡ B) → ua (pathToEquiv e) ≡ e
 ua-pathToEquiv e = uaHAEquiv _ _ .snd .ret e
 
 
 -- Note that for any equivalence (f , e) : X ≃ Y the type  ι (X , s) (Y , t) (f , e) need not to be
 -- a proposition. Indeed this type should correspond to the ways s and t can be identified
 -- as S-structures. This we call a standard notion of structure or SNS.
 -- We will use a different definition, but the two definitions are interchangeable.
-SNS₁ : (S : Type ℓ → Type ℓ') (ι : StrIso S ℓ'') → Type (ℓ-max (ℓ-max (ℓ-suc ℓ) ℓ') ℓ'')
-SNS₁ {ℓ} S ι = ∀ {X : Type ℓ} (s t : S X) → ((s ≡ t) ≃ ι (X , s) (X , t) (idEquiv X))
+SNS₁ : (S : Type ℓ₁ → Type ℓ₂) (ι : StrIso S ℓ₃) → Type (ℓ-max (ℓ-max (ℓ-suc ℓ₁) ℓ₂) ℓ₃)
+SNS₁ {ℓ₁} S ι = ∀ {X : Type ℓ₁} (s t : S X) → ((s ≡ t) ≃ ι (X , s) (X , t) (idEquiv X))
 
 
 -- We introduce the notation for structure preserving equivalences a bit differently,
 -- but this definition doesn't actually change from Escardó's notes.
-_≃[_]_ : (A : TypeWithStr ℓ S) (ι : StrIso S ℓ') (B : TypeWithStr ℓ S) → Type (ℓ-max ℓ ℓ')
+_≃[_]_ : (A : TypeWithStr ℓ₁ S) (ι : StrIso S ℓ₂) (B : TypeWithStr ℓ₁ S) → Type (ℓ-max ℓ₁ ℓ₂)
 A ≃[ ι ] B = Σ[ f ∈ (typ A ≃ typ B) ] (ι A B f)
 
 
--- Our new definition of standard notion of structure SNS₂ using the ⋆ notation.
--- This is easier to work with than SNS₁ wrt Glue-types
-SNS₂ : (S : Type ℓ → Type ℓ') (ι : StrIso S ℓ'') → Type (ℓ-max (ℓ-max (ℓ-suc ℓ) ℓ') ℓ'')
-SNS₂ {ℓ} S ι = (A B : TypeWithStr ℓ S) (e : typ A ≃ typ B)
-             → (equivFun (S ⋆ e) (str A) ≡ (str B)) ≃ (ι A B e)
 
--- We can unfold SNS₂ as follows:
-SNS₃ : (S : Type ℓ → Type ℓ') (ι : StrIso S ℓ'') → Type (ℓ-max (ℓ-max (ℓ-suc ℓ) ℓ') ℓ'')
-SNS₃ {ℓ} S ι = (A B : TypeWithStr ℓ S) (e : typ A ≃ typ B)
-             → (transport (λ i → S (ua e i)) (str A) ≡ (str B)) ≃ (ι A B e)
-
-SNS₂≡SNS₃ : (S : Type ℓ → Type ℓ') (ι : StrIso S ℓ'') → SNS₂ S ι ≡ SNS₃ S ι
-SNS₂≡SNS₃ S ι = refl
+-- The following PathP version of SNS₁ is a bit easier to
+-- work with for the proof of the SIP
+SNS₂ : (S : Type ℓ₁ → Type ℓ₂) (ι : StrIso S ℓ₃) → Type (ℓ-max (ℓ-max (ℓ-suc ℓ₁) ℓ₂) ℓ₃)
+SNS₂ {ℓ₁} S ι = (A B : TypeWithStr ℓ₁ S) (e : typ A ≃ typ B)
+             → (PathP (λ i → S (ua e i)) (str A) (str B)) ≃ (ι A B e)
```

Comment:
Can you flip the equivalence in the conclusion? They way we don't need the `invEquiv` in `sip` and the formulation is more natural. It might make sense to do the same in `SNS₁`.

Reply:
I changed the SNS-names, but I haven't changed the order of the equivalence, as this would break a few things in the proof of the SIP and files in Cubical.Structures. I agree that it would be more natural, but I'm not sure if that is worth the effort. Maybe I could do that in a separate PR?

Reply:
I think this is a good time before we write even more code that relies on this definition. So it would be great if you could do it in this PR.

Shouldn't it be straightforward to implement? You just drop some `invEq` and flip the order in some proofs using equivalence reasoning with `_≃⟨_⟩_`.

Reply:
You're right, better do it now. I hope I can do it today.